bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectSseClient.java

```diff
@@ -0,0 +1,240 @@
+                    .url(apiUrl + "/api/homeappliances/" + haId + "/events").header(ACCEPT, TEXT_EVENT_STREAM).build();
+
+            logger.debugWithHaId(haId, "Create new listener");
+            ServerSentEvent sse = oksse.newServerSentEvent(request, new ServerSentEvent.Listener() {
+
+                @Override
+                public void onOpen(@Nullable ServerSentEvent sse, @Nullable Response response) {
+                    logger.debugWithHaId(haId, "Channel opened");
+                }
+
+                @Override
+                public void onMessage(@Nullable ServerSentEvent sse, @Nullable String id, @Nullable String event,
+                        @Nullable String message) {
+                    if (KEEP_ALIVE.equals(event)) {
+                        logger.debugWithHaId(haId, "KEEP-ALIVE");
+                    } else {
+                        logger.log(Type.DEFAULT, Level.DEBUG, haId, null, Arrays.asList(formatJsonBody(message)), null,
+                                null, "Received id: {}, event: {}", id, event);
+                    }
+
+                    if (message != null && !StringUtils.isEmpty(message) && !EMPTY_EVENT.equals(message)) {
+                        ArrayList<Event> events = mapToEvents(message, haId);
+                        events.forEach(e -> eventListener.onEvent(e));
+                    }
+
+                    if (CONNECTED.equals(event) || DISCONNECTED.equals(event)) {
+                        eventListener.onEvent(new Event(event, null, null));
+                    }
+                }
+
+                @Override
+                public void onComment(@Nullable ServerSentEvent sse, @Nullable String comment) {
+                    logger.debugWithHaId(haId, "Comment received. comment: {}", comment);
+                }
+
+                @Override
+                public boolean onRetryTime(@Nullable ServerSentEvent sse, long milliseconds) {
+                    logger.debugWithHaId(haId, "Retry time {}", milliseconds);
+                    return true; // True to use the new retry time received by SSE
+                }
+
+                @Override
+                public boolean onRetryError(@Nullable ServerSentEvent sse, @Nullable Throwable throwable,
+                        @Nullable Response response) {
+                    boolean retry = true;
+
+                    logger.warnWithHaId(haId, "Error: {}", throwable != null ? throwable.getMessage() : "");
+
+                    if (response != null) {
+                        if (response.code() == HTTP_FORBIDDEN) {
+                            logger.warnWithHaId(haId,
+                                    "Stopping listener! Got FORBIDDEN response from server. Please check if you allowed to access this device.");
+                            retry = false;
+                        } else if (response.code() == HTTP_UNAUTHORIZED) {
+                            logger.errorWithHaId(haId, "Stopping listener! Access token became invalid.");
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please change this to use the correct logging levels and if possible change the ONLINE/OFFLINE status of thing / bridge.

Reply:
Well, to be fair, if your access token becomes (or is invalid) it is a user configuration issue, that has to be changed immediately in order to continue the system functioning, without user interaction inputting valid token, at least in the closed system of Bosch home connect's sense the system is no longer functioning.